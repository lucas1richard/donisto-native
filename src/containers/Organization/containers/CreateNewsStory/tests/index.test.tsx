import * as React from 'react';
import { shallow } from 'enzyme';

import { CreateNewsStory, ICreateNewsStoryProps } from '../index';

describe('<CreateNewsStory />', () => {
  const renderComponent = (props: ICreateNewsStoryProps) => shallow(<CreateNewsStory {...props} />);
  it('Expect to have unit tests specified', () => {
    expect(renderComponent({
      status: 'editing',
      submitStory: jest.fn()
    })).toMatchSnapshot();
  });
  // it('something', () =>);
});
