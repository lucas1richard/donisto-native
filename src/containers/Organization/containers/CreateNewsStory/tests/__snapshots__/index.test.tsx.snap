// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateNewsStory /> Expect to have unit tests specified 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateNewsStory
    status="editing"
    submitStory={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <CreateNewsStoryHeader />,
        <ScrollViewMock>
          <Component
            style={
              Object {
                "padding": 12,
              }
            }
          >
            <ForwardRef
              destroyOnUnmount={true}
              enableReinitialize={false}
              forceUnregisterOnUnmount={false}
              form="CreateNewsStory_FORM_NAME"
              getFormState={[Function]}
              keepDirtyOnReinitialize={false}
              persistentSubmitErrors={false}
              pure={true}
              shouldAsyncValidate={[Function]}
              shouldError={[Function]}
              shouldValidate={[Function]}
              shouldWarn={[Function]}
              submitAsSideEffect={false}
              submitStory={[MockFunction]}
              touchOnBlur={true}
              touchOnChange={false}
              updateUnregisteredFields={false}
            />
          </Component>
        </ScrollViewMock>,
      ],
      "style": Object {
        "height": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Component
            style={
              Object {
                "padding": 12,
              }
            }
          >
            <ForwardRef
              destroyOnUnmount={true}
              enableReinitialize={false}
              forceUnregisterOnUnmount={false}
              form="CreateNewsStory_FORM_NAME"
              getFormState={[Function]}
              keepDirtyOnReinitialize={false}
              persistentSubmitErrors={false}
              pure={true}
              shouldAsyncValidate={[Function]}
              shouldError={[Function]}
              shouldValidate={[Function]}
              shouldWarn={[Function]}
              submitAsSideEffect={false}
              submitStory={[MockFunction]}
              touchOnBlur={true}
              touchOnChange={false}
              updateUnregisteredFields={false}
            />
          </Component>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <ForwardRef
                destroyOnUnmount={true}
                enableReinitialize={false}
                forceUnregisterOnUnmount={false}
                form="CreateNewsStory_FORM_NAME"
                getFormState={[Function]}
                keepDirtyOnReinitialize={false}
                persistentSubmitErrors={false}
                pure={true}
                shouldAsyncValidate={[Function]}
                shouldError={[Function]}
                shouldValidate={[Function]}
                shouldWarn={[Function]}
                submitAsSideEffect={false}
                submitStory={[MockFunction]}
                touchOnBlur={true}
                touchOnChange={false}
                updateUnregisteredFields={false}
              />,
              false,
              false,
            ],
            "style": Object {
              "padding": 12,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "destroyOnUnmount": true,
                "enableReinitialize": false,
                "forceUnregisterOnUnmount": false,
                "form": "CreateNewsStory_FORM_NAME",
                "getFormState": [Function],
                "keepDirtyOnReinitialize": false,
                "persistentSubmitErrors": false,
                "pure": true,
                "shouldAsyncValidate": [Function],
                "shouldError": [Function],
                "shouldValidate": [Function],
                "shouldWarn": [Function],
                "submitAsSideEffect": false,
                "submitStory": [MockFunction],
                "touchOnBlur": true,
                "touchOnChange": false,
                "updateUnregisteredFields": false,
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "destroyOnUnmount": true,
                  "enableReinitialize": false,
                  "forceUnregisterOnUnmount": false,
                  "form": "CreateNewsStory_FORM_NAME",
                  "getFormState": [Function],
                  "keepDirtyOnReinitialize": false,
                  "persistentSubmitErrors": false,
                  "pure": true,
                  "shouldAsyncValidate": [Function],
                  "shouldError": [Function],
                  "shouldValidate": [Function],
                  "shouldWarn": [Function],
                  "submitAsSideEffect": false,
                  "touchOnBlur": true,
                  "touchOnChange": false,
                  "updateUnregisteredFields": false,
                },
                "displayName": "ReduxForm",
                "render": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            false,
            false,
          ],
          "type": [Function],
        },
        "type": [MockFunction],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <CreateNewsStoryHeader />,
          <ScrollViewMock>
            <Component
              style={
                Object {
                  "padding": 12,
                }
              }
            >
              <ForwardRef
                destroyOnUnmount={true}
                enableReinitialize={false}
                forceUnregisterOnUnmount={false}
                form="CreateNewsStory_FORM_NAME"
                getFormState={[Function]}
                keepDirtyOnReinitialize={false}
                persistentSubmitErrors={false}
                pure={true}
                shouldAsyncValidate={[Function]}
                shouldError={[Function]}
                shouldValidate={[Function]}
                shouldWarn={[Function]}
                submitAsSideEffect={false}
                submitStory={[MockFunction]}
                touchOnBlur={true}
                touchOnChange={false}
                updateUnregisteredFields={false}
              />
            </Component>
          </ScrollViewMock>,
        ],
        "style": Object {
          "height": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Component
              style={
                Object {
                  "padding": 12,
                }
              }
            >
              <ForwardRef
                destroyOnUnmount={true}
                enableReinitialize={false}
                forceUnregisterOnUnmount={false}
                form="CreateNewsStory_FORM_NAME"
                getFormState={[Function]}
                keepDirtyOnReinitialize={false}
                persistentSubmitErrors={false}
                pure={true}
                shouldAsyncValidate={[Function]}
                shouldError={[Function]}
                shouldValidate={[Function]}
                shouldWarn={[Function]}
                submitAsSideEffect={false}
                submitStory={[MockFunction]}
                touchOnBlur={true}
                touchOnChange={false}
                updateUnregisteredFields={false}
              />
            </Component>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <ForwardRef
                  destroyOnUnmount={true}
                  enableReinitialize={false}
                  forceUnregisterOnUnmount={false}
                  form="CreateNewsStory_FORM_NAME"
                  getFormState={[Function]}
                  keepDirtyOnReinitialize={false}
                  persistentSubmitErrors={false}
                  pure={true}
                  shouldAsyncValidate={[Function]}
                  shouldError={[Function]}
                  shouldValidate={[Function]}
                  shouldWarn={[Function]}
                  submitAsSideEffect={false}
                  submitStory={[MockFunction]}
                  touchOnBlur={true}
                  touchOnChange={false}
                  updateUnregisteredFields={false}
                />,
                false,
                false,
              ],
              "style": Object {
                "padding": 12,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "destroyOnUnmount": true,
                  "enableReinitialize": false,
                  "forceUnregisterOnUnmount": false,
                  "form": "CreateNewsStory_FORM_NAME",
                  "getFormState": [Function],
                  "keepDirtyOnReinitialize": false,
                  "persistentSubmitErrors": false,
                  "pure": true,
                  "shouldAsyncValidate": [Function],
                  "shouldError": [Function],
                  "shouldValidate": [Function],
                  "shouldWarn": [Function],
                  "submitAsSideEffect": false,
                  "submitStory": [MockFunction],
                  "touchOnBlur": true,
                  "touchOnChange": false,
                  "updateUnregisteredFields": false,
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "destroyOnUnmount": true,
                    "enableReinitialize": false,
                    "forceUnregisterOnUnmount": false,
                    "form": "CreateNewsStory_FORM_NAME",
                    "getFormState": [Function],
                    "keepDirtyOnReinitialize": false,
                    "persistentSubmitErrors": false,
                    "pure": true,
                    "shouldAsyncValidate": [Function],
                    "shouldError": [Function],
                    "shouldValidate": [Function],
                    "shouldWarn": [Function],
                    "submitAsSideEffect": false,
                    "touchOnBlur": true,
                    "touchOnChange": false,
                    "updateUnregisteredFields": false,
                  },
                  "displayName": "ReduxForm",
                  "render": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
              },
              false,
              false,
            ],
            "type": [Function],
          },
          "type": [MockFunction],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
